### PR TITLE
[RD-4115] Update opentelemetry-collector-pano-ec to latest version

### PR DESCRIPTION
**Pull request by skrawn**
This is probably the first time you guys have seen this repo, so let me explain what it is for:

- The main distribution of the [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector) has some default components like the OTLP Receiver, Prometheus Exporter, File Handler Extension etc. but doesn't include things like the Google Cloud Exporter because those are community supported plugins.
- There's another repo where all the community supported plugins live: https://github.com/open-telemetry/opentelemetry-collector-contrib
- OpenTelemetry provides a utility called `builder` that allows you to build your own version of the collector with the components you want.
- Ideally, the collector would be built in the Yocto config (Mender is actually a binary build from Go source) but Yocto has a messy way of handling Go dependencies at the moment.
- So, this repo exists to hold the customized version of the opentelemetry-collector that the EC requires, builds the release binary as part of the release workflow, and then Yocto will grab the tagged binary to install in the target image.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   release:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (agargenta) [inline .github/workflows/release.yaml:11]: not ready to jump to 24.04, huh? :-)

https://github.blog/changelog/2024-05-14-github-hosted-runners-public-beta-of-ubuntu-24-04-is-now-available/

**Review by agargenta — APPROVED** [review]
LGTM

Thanks for the extra context on this repo.

Separately, since this is on the path to production, I suggest adding a branch protection rule that _requires_ PR approvals.

**Review by rigen-panoai — APPROVED** [review]
LGTM

+1 to @agargenta's recommendation on branch protection rules for `main`

File: .github/workflows/release.yaml (hunk 1 above)
Comment (skrawn) [reply .github/workflows/release.yaml:11]: Maybe next update 👍

**Pull request merged**